Expensify/App
# Fix npm version race condition

This sets up a new Github Action that runs a node.js script. If we see the error encountered in the issue, then we'll:

1) Grab the npm patch version from `package.json`
2) Fetch all the tags from the RNC repo
3) Filter out tags from different patch versions
4) Grab the highest build version for which there's a tag in the repo
5) Update the npm version using that number + 1

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/147462

### Tests
**Note:** This is not tested yet, but my plan to test it it:

1) Create two dummy PR's (add some random markdown files)
2) Merge them both in quick succession. Hopefully this should reproduce the original issue
3) Merge this PR into master and wait for it to deploy.
4) Repeat steps 1 and 2, and hopefully the issue should be gone.
